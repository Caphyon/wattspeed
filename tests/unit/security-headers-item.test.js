import { shallowMount } from "@vue/test-utils";
import security_headers_item from "../../src/components/panels/tech/items/security-headers-item";

describe("security-ssl-item.vue", () => {
  it("Snapshot for component level danger", () => {
    const wrapper = shallowMount(security_headers_item, {
      propsData: {
        title: 'Strict-Transport-Security',
        data: {
          value: 'NA',
          set: false,
        },
      },
    });

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.find('div.scrollable--issue.alert--danger').exists()).toBe(true);
    expect(wrapper.find('div.scrollable--issue.alert--success').exists()).toBe(false);
    wrapper.destroy();
  });

  it("Snapshot for component level success", () => {
    const wrapper = shallowMount(security_headers_item, {
      propsData: {
        title: 'Strict-Transport-Security',
        data: {
          value: 'max-age=31536000',
          set: true,
        },
      },
    });

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.find('div.scrollable--issue.alert--danger').exists()).toBe(false);
    expect(wrapper.find('div.scrollable--issue.alert--success').exists()).toBe(true);
    wrapper.destroy();
  });

  it("Snapshot for component level success_noValue", () => {
    const wrapper = shallowMount(security_headers_item, {
      propsData: {
        title: 'X-Frame-Options',
        data: {
          value: 'NA',
          set: true,
        },
      },
    });

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.find('div.scrollable--issue.alert--danger').exists()).toBe(false);
    expect(wrapper.find('div.scrollable--issue.alert--success').exists()).toBe(true);
    wrapper.destroy();
  });

});
