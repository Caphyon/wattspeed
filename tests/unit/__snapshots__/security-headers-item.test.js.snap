// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`security-ssl-item.vue Snapshot for component level danger 1`] = `
<div
  class="scrollable--issue alert--danger"
>
  <h4
    class="mt0 mb0"
  >
    Strict-Transport-Security
  </h4>
   
  <p
    class="mb0"
  >
    <span>
      HSTS tells the browser that the connection should be made using HTTPS, instead of HTTP. 
    </span>
     
    <a
      href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security"
      target="_blank"
    >
      Learn more
    </a>
  </p>
   
  <p
    class="code mb0"
  >
    <code>
      NA
    </code>
  </p>
</div>
`;

exports[`security-ssl-item.vue Snapshot for component level success 1`] = `
<div
  class="scrollable--issue alert--success"
>
  <h4
    class="mt0 mb0"
  >
    Strict-Transport-Security
  </h4>
   
  <p
    class="mb0"
  >
    <span>
      HSTS tells the browser that the connection should be made using HTTPS, instead of HTTP. 
    </span>
     
    <a
      href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security"
      target="_blank"
    >
      Learn more
    </a>
  </p>
   
  <p
    class="code mb0"
  >
    <code>
      max-age=31536000
    </code>
  </p>
</div>
`;

exports[`security-ssl-item.vue Snapshot for component level success_noValue 1`] = `
<div
  class="scrollable--issue alert--success"
>
  <h4
    class="mt0 mb0"
  >
    X-Frame-Options
  </h4>
   
  <p
    class="mb0"
  >
    <span>
      XFO indicate whether or not the browser should be allowed to render a page in a &lt;frame&gt;, &lt;iframe&gt;, &lt;embed&gt; or &lt;object&gt; to avoid click-jacking attacks ensuring that their content is not embedded into other sites. 
    </span>
     
    <a
      href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options"
      target="_blank"
    >
      Learn more
    </a>
  </p>
   
  <p
    class="code mb0"
  >
    <code>
      NA
    </code>
  </p>
</div>
`;
